### PR TITLE
List campaigns

### PR DESCRIPTION
**Pull request by kaiobrito**
### New
- (internal) Networking layer setup
-  List campaigns

### CODE REVIEW

File: plugins/polio/js/src/utils/networking.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const sendRequest = (method, path, body) => {
+    const requestInit = {
+        method,
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        },
+        body: body ? JSON.stringify(body) : undefined,
```
Comment (xavierbarbosa) [inline plugins/polio/js/src/utils/networking.js:8]: The cookie is automatically sent already so you don't need nothing else to authenticate?

**Review by xavierbarbosa — APPROVED** [review]
👌

File: plugins/polio/js/src/utils/networking.js (hunk 1 above)
Comment (kaiobrito) [reply plugins/polio/js/src/utils/networking.js:8]: The cookie is automatically sent. FTR: The `/api/polio/campaign` can't be accessed by non-authenticated users.

File: plugins/polio/js/src/hooks/useCampaign.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { useQuery } from 'react-query';
+import { sendRequest } from '../utils/networking';
+
+export const useCampaign = () => {
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useCampaign.jsx:4]: I think I would call this one 'useGetCampaigns' to make it a bit clearer what it does.

File: plugins/polio/js/src/hooks/useCampaign.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    const {
+        data,
+        isLoading,
+    } = useQuery('polio-campaigns', () => sendRequest('GET', '/api/polio/campaigns/'));
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useCampaign.jsx:8]: Can we change the key to ['polio', 'campaigns']? I find it easier to grasp when it's as close to the API URL as possible

File: plugins/polio/js/src/utils/networking.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export const sendRequest = (method, path, body) => {
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/utils/networking.js:1]: Could we make this an async function?

**Review by geoffreydhuyvetters — APPROVED** [review]

**Pull request merged**